### PR TITLE
customer-wishlist-page

### PR DESCRIPTION
**Pull request by shivendra-webkul**
## Issue Reference
customer wish list page

### CODE REVIEW

File: packages/Webkul/Customer/src/Repositories/WishlistRepository.php [hunk 1]
```diff
@@ -15,72 +15,4 @@ function model(): string
     {
         return 'Webkul\Customer\Contracts\Wishlist';
```
Comment (devansh-webkul) [inline packages/Webkul/Customer/src/Repositories/WishlistRepository.php:16]: Wishlist::class

File: packages/Webkul/Shop/src/Http/Controllers/Customer/WishlistController.php [hunk 2]
```diff
@@ -37,25 +29,21 @@ public function __construct(
...
-            'items'              => $this->wishlistRepository->getCustomerWishlist(),
-            'isSharingEnabled'   => $this->isSharingEnabled(),
-            'isWishlistShared'   => $customer->isWishlistShared(),
-            'wishlistSharedLink' => $customer->getWishlistSharedLink(),
-        ]);
+        $items = $this->wishlistRepository
+            ->where([
+                'channel_id'=>core()->getCurrentChannel()->id,
+                'customer_id'=>auth()->guard('customer')->user()->id,
+            ])
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/Customer/WishlistController.php:43]: Spacing and indentation.

File: packages/Webkul/Shop/src/Resources/lang/en/app.php [hunk 3]
```diff
@@ -446,6 +446,10 @@
             ],
 
             'wishlist' => [
+                'color'                => 'Color:',
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/lang/en/app.php:449]: Remove  `:`.

File: packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php [hunk 4]
```diff
@@ -1,67 +1,69 @@
...
 
-            @if ($visibility ?? false)
-                <div class="mb-2">
-                    <span class="fs16">
-                        {{ __('shop::app.customer.account.wishlist.visibility') }} :
+            <x-shop::form
+                id="wishlist-{{ $item->id }}" 
+                action="{{ route('shop.customer.wishlist.remove', $item->id) }}" 
+            >
+                @method('DELETE')
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php:32]: No need to add DELETE here. Just pass `method="DELETE"` in x-form.

File: packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php [hunk 5]
```diff
@@ -1,67 +1,69 @@
...
-            {{ __('shop::app.customer.account.wishlist.move-to-cart') }}
-        </a>
+        <span class="bg-[position:-172px_-44px] bs-main-sprite w-[14px] h-[14px]"></span>
     </div>
-</div>
 
-<div class="horizontal-rule mb-10 mt-10"></div>
\ No newline at end of file
+    <a href="{{ route('shop.customer.wishlist.move', $item->id) }}"> 
+
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php:62]: Remove line.

File: packages/Webkul/Shop/src/Routes/customer-routes.php [hunk 6]
```diff
@@ -141,9 +141,7 @@
                     /**
                      * Wishlist.
                      */
-                    Route::get('wishlist', [WishlistController::class, 'index'])->defaults('_config', [
-                        'view' => 'shop::customers.account.wishlist.wishlist',
-                    ])->name('shop.customer.wishlist.index');
+                    Route::get('wishlist', [WishlistController::class, 'index'])->name('shop.customer.wishlist.index');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/customer-routes.php:144]: shop.customers.account.wishlist.index

Fix all the wishlist routes also. Consult with Amit.

**Review by devansh-webkul — COMMENTED** [review]
Please check the above changes.

**Review by devansh-webkul — COMMENTED** [review]
Please check the above changes.

File: packages/Webkul/Shop/src/Http/Controllers/Customer/WishlistController.php [hunk 7]
```diff
@@ -37,25 +29,21 @@ public function __construct(
...
-            'wishlistSharedLink' => $customer->getWishlistSharedLink(),
-        ]);
+        $items = $this->wishlistRepository
+            ->where([
+                'channel_id'  => core()->getCurrentChannel()->id,
+                'customer_id' => auth()->guard('customer')->user()->id,
+            ])
+            ->paginate(5);
+
+        return view('shop::customers.account.wishlist.wishlist', compact('items'));
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/Customer/WishlistController.php:46]: 'shop::customers.account.wishlist.index'

File: packages/Webkul/Customer/src/Repositories/WishlistRepository.php (hunk 1 above)
Comment (devansh-webkul) [reply packages/Webkul/Customer/src/Repositories/WishlistRepository.php]: Fixed

File: packages/Webkul/Shop/src/Resources/lang/en/app.php (hunk 3 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Resources/lang/en/app.php]: done

File: packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php (hunk 5 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php]: Done

File: packages/Webkul/Shop/src/Routes/customer-routes.php (hunk 6 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Routes/customer-routes.php]: Done

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php [hunk 8]
```diff
@@ -32,51 +32,51 @@ public function run()
                     <ul class="grid gap-[20px] text-[14px]">
                         <li>
                             <a href="' . route('shop.cms.page', 'about-us') . '">
-                                ' . trans('shop::app.footer.about-us') . '
+                                ' . trans('shop::app.componets.layouts.footer.about-us') . '
```
Comment (devansh-webkul) [inline packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php:35]: Typo.

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php [hunk 9]
```diff
@@ -32,51 +32,51 @@ public function run()
...
                             </a>
                         </li>
 
                         <li>
                             <a href="' . route('shop.cms.page', 'customer-service') . '">
-                                ' . trans('shop::app.footer.customer-service') . '
+                                ' . trans('shop::app.componets.layouts.footer.customer-service') . '
```
Comment (devansh-webkul) [inline packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php:41]: Same here.

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php [hunk 10]
```diff
@@ -32,51 +32,51 @@ public function run()
...
                             </a>
                         </li>
 
                         <li>
                             <a href="' . route('shop.cms.page', 'whats-new') . '">
-                                ' . trans('shop::app.footer.whats-new') . '
+                                ' . trans('shop::app.componets.layouts.footer.whats-new') . '
```
Comment (devansh-webkul) [inline packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php:47]: Same here.

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php [hunk 11]
```diff
@@ -32,51 +32,51 @@ public function run()
...
                             </a>
                         </li>
 
                         <li>
                             <a href="' . route('shop.cms.page', 'contact-us') . '">
-                                ' . trans('shop::app.footer.contact-us') . '
+                                ' . trans('shop::app.componets.layouts.footer.contact-us') . '
```
Comment (devansh-webkul) [inline packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php:53]: Same here.

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php [hunk 12]
```diff
@@ -32,51 +32,51 @@ public function run()
...
                             </a>
                         </li>
                     </ul>
 
                     <ul class="grid gap-[20px] text-[14px]">
                         <li>
                             <a href="' . route('shop.cms.page', 'order-return') . '">
-                                ' . trans('shop::app.footer.order-return') . '
+                                ' . trans('shop::app.componets.layouts.footer.order-return') . '
```
Comment (devansh-webkul) [inline packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php:61]: Same here.

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php [hunk 13]
```diff
@@ -32,51 +32,51 @@ public function run()
...
                             </a>
                         </li>
 
                         <li>
                             <a href="' . route('shop.cms.page', 'payment-policy') . '">
-                                ' . trans('shop::app.footer.payment-policy') . '
+                                ' . trans('shop::app.componets.layouts.footer.payment-policy') . '
```
Comment (devansh-webkul) [inline packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php:67]: Same here.

File: packages/Webkul/Shop/src/Routes/customer-routes.php [hunk 14]
```diff
@@ -73,24 +73,17 @@
...
-                Route::get('wishlist/shared', [WishlistController::class, 'shared'])
-                    ->defaults('_config', [
-                        'view' => 'shop::customers.account.wishlist.wishlist-shared',
-                    ])
-                    ->withoutMiddleware('customer')
-                    ->name('shop.customer.wishlist.shared');
+                Route::post('wishlist', [WishlistController::class, 'store'])->name('shop.customers.account.wishlist.add');
 
-                Route::delete('wishlist/remove/{id}', [WishlistController::class, 'remove'])->name('shop.customer.wishlist.remove');
+                Route::get('wishlist/move-to-cart/{id}', [WishlistController::class, 'moveToCart'])->name('shop.customers.account.wishlist.move_to_cart');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/customer-routes.php:82]: This should be a post request.

`POST wishlist/{id}/move-to-cart`

File: packages/Webkul/Shop/src/Routes/customer-routes.php [hunk 15]
```diff
@@ -73,24 +73,17 @@
...
 
-                Route::delete('wishlist/remove-all', [WishlistController::class, 'removeAll'])->name('shop.customer.wishlist.remove_all');
+                Route::delete('wishlist/{id}', [WishlistController::class, 'remove'])->name('shop.customers.account.wishlist.remove');
 
-                Route::get('wishlist/move/{id}', [WishlistController::class, 'move'])->name('shop.customer.wishlist.move');
+                Route::post('wishlist/mass-delete', [WishlistController::class, 'removeAll'])->name('shop.customers.account.wishlist.remove_all');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/customer-routes.php:86]: `DELETE wishlist/mass-destroy`.

File: packages/Webkul/Shop/src/Routes/customer-routes.php [hunk 16]
```diff
@@ -73,24 +73,17 @@
...
+                */
+                Route::get('wishlist', [WishlistController::class, 'index'])->name('shop.customers.account.wishlist.index');
 
-                Route::get('wishlist/shared', [WishlistController::class, 'shared'])
-                    ->defaults('_config', [
-                        'view' => 'shop::customers.account.wishlist.wishlist-shared',
-                    ])
-                    ->withoutMiddleware('customer')
-                    ->name('shop.customer.wishlist.shared');
+                Route::post('wishlist', [WishlistController::class, 'store'])->name('shop.customers.account.wishlist.add');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/customer-routes.php:80]: shop.customers.account.wishlist.store

File: packages/Webkul/Shop/src/Routes/customer-routes.php [hunk 17]
```diff
@@ -73,24 +73,17 @@
...
 
-                Route::delete('wishlist/remove/{id}', [WishlistController::class, 'remove'])->name('shop.customer.wishlist.remove');
+                Route::get('wishlist/move-to-cart/{id}', [WishlistController::class, 'moveToCart'])->name('shop.customers.account.wishlist.move_to_cart');
 
-                Route::delete('wishlist/remove-all', [WishlistController::class, 'removeAll'])->name('shop.customer.wishlist.remove_all');
+                Route::delete('wishlist/{id}', [WishlistController::class, 'remove'])->name('shop.customers.account.wishlist.remove');
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Routes/customer-routes.php:84]: shop.customers.account.wishlist.destroy

File: packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php [hunk 18]
```diff
@@ -1,67 +1,67 @@
-<div class="account-item-card mt-15 mb-15">
-    <div class="media-info">
+<div class="flex gap-[65px] p-[25px] items-center border-b-[1px] border-[#E9E9E9]">
+    <div class="flex gap-x-[20px]">
+
         @php
             $image = $item->product->getTypeInstance()->getBaseImage($item);
+            $color = app('Webkul\Attribute\Repositories\attributeOptionRepository')->findOrFail($item->product->color);
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php:6]: Remove this. Currently at this stage just leave it as it is.

File: packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/index.blade.php [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+            <p class="flex items-center gap-x-[4px] text-[#7D7D7D] text-[16px] after:content-['/'] after:last:hidden">
+                @lang('shop::app.customers.account.wishlist.title')
+            </p>
+        </div>
+
+        <h2 class="text-[26px] font-medium">
+            @lang('shop::app.customers.account.wishlist.page-title')
+        </h2>
+
+        <!-- wishlist Products -->
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/index.blade.php:17]: single line comment
{{-- wishlist producs --}}

multiline comment
{{--
    Wishlist Products:
    This is a wishlist product.
--}}

**Review by devansh-webkul — CHANGES_REQUESTED** [review]
Check the above changes.

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php (hunk 8 above)
Comment (shivendra-webkul) [reply packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php]: Done

File: packages/Webkul/Shop/src/Routes/customer-routes.php (hunk 14 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Routes/customer-routes.php]: Done

File: packages/Webkul/Shop/src/Routes/customer-routes.php (hunk 15 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Routes/customer-routes.php]: Done

File: packages/Webkul/Shop/src/Routes/customer-routes.php (hunk 16 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Routes/customer-routes.php]: Done

File: packages/Webkul/Shop/src/Routes/customer-routes.php (hunk 17 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Routes/customer-routes.php]: Done

File: packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php (hunk 18 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/wishlist-product.blade.php]: Done

File: packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/index.blade.php (hunk 19 above)
Comment (shivendra-webkul) [reply packages/Webkul/Shop/src/Resources/views/customers/account/wishlist/index.blade.php]: Done

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php (hunk 9 above)
Comment (shivendra-webkul) [reply packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php]: Done

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php (hunk 10 above)
Comment (shivendra-webkul) [reply packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php]: Done

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php (hunk 11 above)
Comment (shivendra-webkul) [reply packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php]: Done

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php (hunk 12 above)
Comment (shivendra-webkul) [reply packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php]: Done

File: packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php (hunk 13 above)
Comment (shivendra-webkul) [reply packages/Webkul/Core/src/Database/Seeders/ChannelTableSeeder.php]: Done

File: packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php [hunk 20]
```diff
@@ -29,7 +29,7 @@
 
             <div class="list-container">
                 @if(core()->getConfigData('customer.settings.newsletter.subscription'))
-                    <label class="list-heading" for="subscribe-field">{{ __('shop::app.footer.subscribe-newsletter') }}</label>
+                    <label class="list-heading" for="subscribe-field">{{ __('shop::app.components.layouts.footer.subscribe-newsletter') }}</label>
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php:32]: @lang()

File: packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php [hunk 21]
```diff
@@ -49,7 +49,7 @@
                     }
                 ?>
 
-                <label class="list-heading" for="locale-switcher">{{ __('shop::app.footer.locale') }}</label>
+                <label class="list-heading" for="locale-switcher">{{ __('shop::app.components.layouts.footer.locale') }}</label>
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php:52]: @lang()

File: packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php [hunk 22]
```diff
@@ -67,7 +67,7 @@
                 </div>
 
                 <div class="currency">
-                    <label class="list-heading" for="currency-switcher">{{ __('shop::app.footer.currency') }}</label>
+                    <label class="list-heading" for="currency-switcher">{{ __('shop::app.components.layouts.footer.currency') }}</label>
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/layouts/footer/footer.blade.php:70]: @lang()

**Review by devansh-webkul — APPROVED** [review]

**Review by jitendra-webkul — APPROVED** [review]

**Pull request merged**